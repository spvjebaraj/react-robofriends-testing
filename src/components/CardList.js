import React from "react";
import Card from "./Card";

const CardList = ({ robots }) => {
  // if (true) {
  //   throw new Error("Something went wrong!");
  // }
  return (
    <React.Fragment>
      {robots.map((user, idx) => (
        <Card key={idx} id={user.id} name={user.name} email={user.email} />
      ))}
    </React.Fragment>
  );
};

export default CardList;
