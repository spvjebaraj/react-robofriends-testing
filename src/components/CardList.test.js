import React from "react";
import { shallow } from "enzyme";
import CardList from "./CardList";

const mockRobots = [
  {
    id: 1,
    name: "Philip",
    username: "philiphilip",
    email: "philip@gmail.com",
  },
  {
    id: 2,
    name: "Anitha",
    username: "anitha",
    email: "anitha@gmail.com",
  },
];

it("expect to render CardList compoent", () => {
  expect(shallow(<CardList robots={mockRobots} />).length).toEqual(1);
});

it("expect to render CardList compoent - Snapshot", () => {
  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
