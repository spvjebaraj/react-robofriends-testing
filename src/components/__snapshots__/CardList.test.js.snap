// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList compoent - Snapshot 1`] = `ShallowWrapper {}`;
